core/unit_test/TestSharedHostPinnedSpace.cpp

```diff
@@ -0,0 +1,163 @@
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+#include <gtest/gtest.h>
+#include <Kokkos_Core.hpp>
+
+namespace {
+
+template <typename ViewType>
+class IncrementFunctor {
+ private:
+  using index_type = decltype(std::declval<ViewType>().size());
+  ViewType view_;
+
+ public:
+  IncrementFunctor() = delete;
+
+  explicit IncrementFunctor(ViewType view) : view_(view) {}
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const index_type idx) const { ++view_(idx); }
+};
+
+template <typename ExecSpace, typename ViewType>
+void incrementInLoop(ViewType& view) {
+  using index_type = decltype(view.size());
+
+  Kokkos::fence();
+  IncrementFunctor<ViewType> func(view);
+  Kokkos::parallel_for(
+      "increment",
+      Kokkos::RangePolicy<ExecSpace, Kokkos::IndexType<index_type>>{
+          0, view.size()},
+      func);
+  Kokkos::fence();
+}
+
+template <typename ViewType>
+class CheckFunctor {
+ private:
+  using index_type = decltype(std::declval<ViewType>().size());
+  ViewType view_;
+  const int targetVal_;
+
+ public:
+  CheckFunctor() = delete;
+
+  CheckFunctor(ViewType view, int targetVal)
+      : view_(view), targetVal_(targetVal) {}
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const index_type idx, bool& equality) const {
+    (view_(idx) == targetVal_) ? equality &= true : equality &= false;
```

Comment:
Yuk yuk yuk
Prefer `bool& equality -> int& err`
and
```C++
if (view_(idx) != targetVal_) { ++err; }

Reply:
changed in [ab0f14](https://github.com/kokkos/kokkos/pull/5405/commits/ab0f14be7e2cc2777de17625a9b32f9b40e7f3b1)